helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/AvroSchemaUtil.java

```diff
@@ -117,6 +174,159 @@ public static Schema findNonNullUnionBranch(Schema schema) {
     return nonNullBranches.get(0);
   }
 
+  /**
+   * implements "schema resolution", as outlined in the relevant section of (recent-ish) the avro specification
+   * @param writer
+   * @param reader
+   * @param useAliases
+   * @return
+   */
+  public static SchemaResolutionResult resolveReaderVsWriter(Schema writer, Schema reader, boolean useAliases, boolean usePromotions) {
+    Schema.Type writerType = writer.getType();
+    Schema.Type readerType = reader.getType();
+
+    if (writerType == Schema.Type.UNION) {
+      if (readerType == Schema.Type.UNION) {
+        //if both are unions: The first schema in the reader's union that matches the selected writer's union schema
+        //is recursively resolved against it. if none match, an error is signalled.
+        //1st attempt is strict
```

Comment:
I don't think I follow...why is do we only resolve against the first match?

Reply:
im quoting the spec here. will clarify that